CRM/Core/Payment.php

```diff
@@ -149,6 +149,338 @@ abstract class CRM_Core_Payment {
+    $this->contactID = $contactID;
+  }
+
+  /**
+   * @return int
+   */
+  public function getContributionRecurID(): int {
+    return $this->contributionRecurID ?? $this->inputParams['contributionRecurID'];
+  }
+
+  /**
+   * @param int $contributionRecurID
+   */
+  public function setContributionRecurID(int $contributionRecurID) {
+    $this->contributionRecurID = $contributionRecurID;
+  }
+
+  /**
+   * @return string
+   */
+  public function getPaymentToken(): string {
+    return $this->paymentToken ?? $this->inputParams['token'];
+  }
+
+  /**
+   * @param string $paymentToken
+   */
+  public function setPaymentToken(string $paymentToken) {
+    $this->paymentToken = $paymentToken;
+  }
+
+  /**
+   * Get gateway generated transaction ID.
+   *
+   * @return string
+   */
+  public function getTransactionID(): string {
+    return $this->transactionID;
+  }
+
+  /**
+   * Set gateway generated transaction ID.
+   *
+   * @param string $transactionID
+   */
+  public function setTransactionID(string $transactionID) {
+    $this->transactionID = $transactionID;
+  }
+
+  /**
+   * Get additional result information received from gateway.
+   *
+   * This is saved to civicrm_financial_trxn.result_code.
+   *
+   * @return string
+   */
+  public function getResultCode(): string {
+    return $this->resultCode;
+  }
+
+  /**
+   * Set additional result information from gateway.
+   *
+   * This is saved to civicrm_financial_trxn.result_code.
+   *
+   * @param string $resultCode
+   */
+  public function setResultCode(string $resultCode) {
+    $this->resultCode = $resultCode;
+  }
+
+  /**
+   * Get recurring frequency interval.
+   *
+   * @return int
+   */
+  public function getRecurFrequencyInterval(): int {
+    return $this->recurFrequencyInterval ?? $this->inputParams['frequency_interval'];
+  }
+
+  /**
+   * Set recurring frequency interval.
+   *
+   * @param int $recurFrequencyInterval
+   */
+  public function setRecurFrequencyInterval(int $recurFrequencyInterval) {
+    $this->recurFrequencyInterval = $recurFrequencyInterval;
+  }
+
+  /**
+   * Get recurring frequency unit.
+   *
+   * @return string
+   */
+  public function getRecurFrequencyUnit(): string {
+    return $this->recurFrequencyUnit;
+  }
+
+  /**
+   * Set recurring frequency unit.
+   *
+   * @param string $recurFrequencyUnit
+   */
+  public function setRecurFrequencyUnit(string $recurFrequencyUnit) {
+    $this->recurFrequencyUnit = $recurFrequencyUnit ?? $this->inputParams['frequency_unit'];
+  }
+
+  /**
+   * Get invoice ID (CiviCRM generated invoice reference).
+   *
+   * @return string
+   */
+  public function getInvoiceID(): string {
+    return $this->invoiceID ?? $this->inputParams['invoiceID'];
+  }
+
+  /**
+   * Set invoice ID (CiviCRM generated invoice reference).
+   *
+   * @param string $invoiceID
+   */
+  public function setInvoiceID(string $invoiceID) {
+    $this->invoiceID = $invoiceID;
+  }
```

Comment:
Yes. Nice and clear. Maybe later we'd move code that generates the invoice ID to `getInvoiceID` so we don't have to sprinkle it through various API / BAO functions.

Reply:
I think it should be in Order.create - I think https://lab.civicrm.org/dev/financial/issues/78

Reply:
I've commented on that issue.